packages/apps/desktop-viewer-test/src/frontend/extensions/IModelMergeStatusBarItem.tsx

```diff
@@ -0,0 +1,122 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import "./IModelMergeStatusBarItem.scss";
+
+import { InternetConnectivityStatus } from "@bentley/imodeljs-common";
+import { BriefcaseConnection, IModelApp } from "@bentley/imodeljs-frontend";
+import {
+  CommonStatusBarItem,
+  StageUsage,
+  StatusBarSection,
+  UiItemsProvider,
+} from "@bentley/ui-abstract";
+import { StatusBarItemUtilities, UiFramework } from "@bentley/ui-framework";
+import {
+  getBriefcaseStatus,
+  ModelStatus,
+  useAccessToken,
+  useConnectivity,
+} from "@itwin/desktop-viewer-react";
+import { SvgUser } from "@itwin/itwinui-icons-react";
+import React, { useCallback, useEffect, useState } from "react";
+
+import { BriefcaseStatus } from "../components/modelSelector";
+
+const MergeStatusBarItem = () => {
+  const [mergeStatus, setMergeStatus] = useState<ModelStatus>();
+  const [connection, setConnection] = useState<BriefcaseConnection>();
+  const accessToken = useAccessToken();
+  const connectivityStatus = useConnectivity();
+
+  const onMergeClick = async () => {
+    setMergeStatus(ModelStatus.MERGING);
+  };
+
+  const onLoginClick = async () => {
+    await IModelApp.authorizationClient?.signIn();
+  };
+
+  const getLatestChangesets = useCallback(async () => {
+    if (connection) {
+      try {
+        await connection.pullAndMergeChanges();
+        setMergeStatus(ModelStatus.UPTODATE);
+        IModelApp.viewManager.refreshForModifiedModels(undefined);
```

Comment:
Is this actually even possible? Merging a model while its open? Ive had no lcukw ith it so far, fails to apply changesets.

The sqlite file is open for read, so how would it apply/write any new changes?

Reply:
the briefcase connection is opened in readonly state (see openLocalIModel in IModelService), which means you can "pull" changes, but not push. You definitely cannot merge changes back to the model. That is out of scope here. Merge maybe isn't the best word. Maybe just pulling?

What are you attempting here? The proper way to test this is to open a model, then sync new changes to the model, and then click merge. The new changes should be evident in the local briefcase. I havent had any issues with that workflow so let me know if I am missing something

Reply:
Downloaded model
Sync changes via demo portal
Restarted app, opened the downloaded model before merging
The status bar item lets me merge (which i dont think it should) so i click it, fails to merge/pull
![image](https://user-images.githubusercontent.com/11051042/144503692-52469055-68fc-4370-a198-b2b6ee6356a8.png)
![image](https://user-images.githubusercontent.com/11051042/144503914-9f4f18f5-774c-4f99-884f-c15b4e962212.png)
![image](https://user-images.githubusercontent.com/11051042/144503928-9707e4cc-f65b-4e97-9922-90062fca680e.png)

Fails to close the bim file when its open in the viewer, which makes sense.  I dont think it should actually should getLatestChnageset when youre in the viewer, just alert the user to go back and merge from the Select pg again

Reply:
I then navigated back to the imodel selection pg (which i wouldve assumed closed the connection), and then tried merging  again but that failed with the same applying changeset error.

I opened another model, went back to merge again still failed. Not sure whats doing on. The only error i see in the log is BE_SQLITE_BUSY

Reply:
Restarting the viewer, and merging from the select pg without ever opening the model is fine, but if you open it and then try merging it will def fail in the native layer when trying to apply the changesets

Reply:
Ok thanks.....I havent been able to replicate this so I'm going to attempt on Windows.

Reply:
As discussed, this appears to be a bug in core. Essentially the iModel DB allows pulling changes in a read-only connection. To do so, it temporarily re-opens as writable, applies changes, and then attempts to re-open as readonly again. The last part is what is failing. It appears to have been addressed in 3.0, so will re-test again once merged into next. In the meantime, I made some other changes that will hopefully help with displaying the changed model in the viewport. Please verify and let me know if that still does not occur

Reply:
Working for me now.